Port MASTG-TEST-0054: Determining Whether Sensitive Data Is Shared with Third Parties (ios) (by @guardsquare) (#3047)
OWASP/owasp-mastg: tests-beta/ios/MASVS-STORAGE/MASTG-TEST-0x54.md

---
platform: ios

> title: Sending data to trackers' domains

Reviewer: ```suggestion
title: Undeclared Known Tracking Domains
```

> id: MASTG-TEST-0x54

Reviewer: ```suggestion
id: MASTG-TEST-0281
```

type: [static, dynamic]
weakness: MASWE-0108
---

## Overview

This test checks if your app connects to tracker services that could monitor user behavior and display banner ads.

## Steps

1. Collect a list of potential trackers. You can use an open-source list like [lists](https://github.com/duckduckgo/tracker-blocklists/blob/main/app/android-tds.json).

2. Search statically with @MASTG-TOOL-0110 for deny listed domain names, or dynamically intercept network requests with @MASTG-TOOL-0097.

3. Review the list of used domains.

## Observation

> The output should contain the list of:
> 
> - matched domains with the deny list (static analysis)
> - all the domains that the app interacted with (dynamic analysis)

Reviewer: ```suggestion
The output should contain:

- a list of tracking domains with which the app has interacted, or may interact.
- all the app's privacy manifests as files.
```
